model/lightgbm/dffml_model_lightgbm/lgbmclassifier.py

```diff
@@ -0,0 +1,251 @@
+    properties={
+        "features": (Features, field("Features to train on")),
+        "predict": (Feature, field("Label or the value to be predicted")),
+        "directory": (
+            pathlib.Path,
+            field("Directory where state should be saved",),
+        ),
+    },
+)
+
+
+@entrypoint("lgbmclassifier")
+class LGBMClassifierModel(SimpleModel):
+    r"""
+    LightGBM (Light Gradient Boosting Machine) is a free and open source distributed gradient boosting framework for machine learning originally developed by Microsoft. 
+    It is much faster to train and based on decision tree algorithms and used for ranking, classification and other machine learning tasks.
+    This model is used for classification problems.
+    Examples
+    ----------------------
+
+    Command line usage
+
+    Download the training and testing date sets, change the headers to DFFML format.
+
+    .. code-block::
+        :test:
+
+        $ wget http://download.tensorflow.org/data/iris_training.csv
+        $ wget http://download.tensorflow.org/data/iris_test.csv
+        $ sed -i 's/.*setosa,versicolor,virginica/SepalLength,SepalWidth,PetalLength,PetalWidth,classification/g' iris_training.csv iris_test.csv
+
+
+    Run the train command
+
+    .. code-block:: console
+        :test:
+
+        $ dffml train \
+            -sources train=csv \
+            -source-filename iris_training.csv \
+            -model lgbmclassifier \
+            -model-features \
+              SepalLength:float:1 \
+              SepalWidth:float:1 \
+              PetalLength:float:1 \
+              PetalWidth:float:1 \
+            -model-predict classification \
+            -model-directory model \
+            -model-max_depth 4 \
+            -model-learning_rate 0.1 \
+            -model-n_estimators 100 \
+            -model-reg_lambda 1 \
+            -model-reg_alpha 0 \
+            -model-subsample 1 
+
+
+    Assess the accuracy 
+
+    .. code-block:: console
+        :test:
+
+        $ dffml accuracy \
+            -sources train=csv \
+            -source-filename iris_test.csv \
+            -model lgbmclassifier \
+            -model-features \
+              SepalLength:float:1 \
+              SepalWidth:float:1 \
+              PetalLength:float:1 \
+              PetalWidth:float:1 \
+            -model-predict classification \
+            -model-directory model 
+
+    Make predictions
+
+    .. code-block:: console
+        :test:
+
+        $ dffml predict all \
+            -sources train=csv \
+            -source-filename iris_test.csv \
+            -model lgbmclassifier \
+            -model-features \
+              SepalLength:float:1 \
+              SepalWidth:float:1 \
+              PetalLength:float:1 \
+              PetalWidth:float:1 \
+            -model-predict classification \
+            -model-directory model 
+
+    
+    Python usage
+
+    .. literalinclude:: /../model/lightgbm/examples/iris_classification.py
```

Comment:
Let's make sure to test this
```suggestion
    .. literalinclude:: /../model/lightgbm/examples/iris_classification.py
        :test:
```